cppseminar/ApcScaler/ApcScaler/Workers.cs

```diff
@@ -0,0 +1,133 @@
+﻿
+using Microsoft.Azure.Management.Compute.Fluent;
+using Microsoft.Azure.Management.Fluent;
+using Microsoft.Azure.Management.ResourceManager.Fluent.Authentication;
+using Microsoft.Azure.Management.ResourceManager.Fluent;
+
+namespace ApcScaler
+{
+    public class Workers
+    {
+        private readonly ILogger<Scaler> _logger;
+        private readonly IConfiguration _configuration;
+
+        private readonly string _apcWorkerResourceGroup;
+        private readonly string _apcWorkerVmNamePrefix;
+
+        private readonly IAzure? _azureClient;
+        public Workers(ILogger<Scaler> logger, IConfiguration configuration)
+        {
+            _logger = logger;
+            _configuration = configuration;
+
+            _apcWorkerResourceGroup = _configuration["APC_WORKER_RESOURCE_GROUP"];
+            _apcWorkerVmNamePrefix = _configuration["APC_WORKER_VM_NAME_PREFIX"];
+
+            _ = !string.IsNullOrWhiteSpace(_apcWorkerResourceGroup) ? _apcWorkerResourceGroup : 
+                throw new Exception("Missing configuration value of [APC_WORKER_RESOURCE_GROUP]");
+            _ = !string.IsNullOrWhiteSpace(_apcWorkerVmNamePrefix) ? _apcWorkerVmNamePrefix : 
+                throw new Exception("Missing configuration value of [APC_WORKER_VM_NAME_PREFIX]");
+
+            var subscriptionId = _configuration["AZURE_SUBSCRIPTION_ID"];
+            var tenantId = _configuration["AZURE_TENANT_ID"];
+            var clientId = _configuration["AZURE_CLIENT_ID"];
+            var clientSecret = _configuration["AZURE_CLIENT_SECRET"];
+
+            _ = !string.IsNullOrWhiteSpace(subscriptionId) ? subscriptionId :
+                throw new Exception("Missing configuration value of [AZURE_SUBSCRIPTION_ID]");
+            _ = !string.IsNullOrWhiteSpace(tenantId) ? tenantId :
+                throw new Exception("Missing configuration value of [AZURE_TENANT_ID]");
+            _ = !string.IsNullOrWhiteSpace(clientId) ? clientId :
+                throw new Exception("Missing configuration value of [AZURE_CLIENT_ID]");
+            _ = !string.IsNullOrWhiteSpace(clientSecret) ? clientSecret :
+                throw new Exception("Missing configuration value of [AZURE_CLIENT_SECRET]");
+
+            try
+            {
+                var creds = new AzureCredentialsFactory().FromServicePrincipal(clientId,
+                    clientSecret, tenantId, AzureEnvironment.AzureGlobalCloud);
+
+                _azureClient = Azure.Authenticate(creds).WithSubscription(subscriptionId);
+
+                _logger.LogDebug("Azure client successfully created");
+            }
+            catch (Exception e)
```

Comment:
This I do not understand. Maybe C# is different, but why do we catching exceptions in constructor, that make no sense to me. Because this way we will have invalid object (_azureClient will by null), so why ?